scripts/react_native_pods.rb

```diff
@@ -149,7 +149,7 @@ def use_react_native_codegen!(spec, options={})
   return if ENV['DISABLE_CODEGEN'] == '1'
 
   # The path to react-native (e.g. react_native_path)
-  prefix = options[:path] ||= File.join(__dir__, "..")
+  prefix = options[:path] ||= "../../node_modules/react-native"
```

Comment:
This is the correct path for the script_phase but not for `prepare_command`, `prepare_command` is relative to the  podspec, and not `PODS_ROOT` like `script_phase`. Not sure how this is working for you, All your generated code should be in the wrong directory. It might work but is not the expected result. The generated code should exist under `node_modules/react-native/React/`

Reply:
Thank you @martintreurnicht, you are correct. The prepare command was run for the wrong path; after running `pod install` an extra `node_modules` directory was created under `node_modules/react-native`.

The was working for me because apparently the prepare command is a bit redundant; after checking out react-native the empty files are already there so just the timestamps were not updated.

I've fixed the script now. The end result is a hack that I do not like, but at least it should work now. If someone knows better how the Pod build works and how this problem should be properly addressed please let me know (or make another PR and this one can be closed).

For reference (TIL): the `prepare_command` is run with `pod install`; you can check the working directory by adding `pwd && exit 1` at the beginning of it (spoiler: it's `/local/dev/path/node_modules/react-native/React/FBReactNativeSpec`). Also, the `script_phase` is run with the actual app build (`yarn run-ios`).

Reply:
First off, thanks for bringing this issue to my attention! I agree this needs to be fixed.
I am not sure at this moment that hard-coding this prefix to `"../../node_modules/react-native"` will work in every scenario. While this path is valid for standalone apps that consume `react-native` from npm, the path will not exist if this script is invoked when `pod install` is run within `packages/rn-tester` (e.g. when building RNTester as a contributor to React Native).